wizard import fatturapa: variabile sovrascritta se ci sono DatiFattureCollegate

se ci sono variabile sovrascritta se ci sono DatiFattureCollegate:

alla linea 968 di wizard_import_fatturapa.py
```python
for invoice in RelInvoices:
```
ma sovrascrive invoce di qualche riga sopra di tipo `account.invoice`, mentre diventa `DatiDocumentiCorrelatiType`

poi fallisce con:
```
  File "/opt/odoo/10.0/ce/prod/addons/l10n_it_fatturapa_in/wizard/wizard_import_fatturapa.py", line 1046, in invoiceCreate
    invoice.write(delivery_dict)
AttributeError: 'DatiDocumentiCorrelatiType' object has no attribute 'write'
```

ho sostituito con
```python
            for rel_invoice in RelInvoices:
                doc_datas = self._prepareRelDocsLine(
                    invoice_id, rel_invoice, 'invoice')
                if doc_datas:
                    for doc_data in doc_datas:
                        rel_docs_model.create(doc_data)
```

sostituendo il parametro in _prepareRelDocsLine rel_invoice ma non so se basta/è corretto non avendo guardato la logica

al momento la branch 10.0 (oca) mi fallisce 4 test, ho rotto qualcosa io?

```
FAIL: test_04_xml_import (odoo.addons.l10n_it_fatturapa_in.tests.test_import_fatturapa_xml.TestFatturaPAXMLValidation)
: Traceback (most recent call last):
: `   File "/home/sherpya/workspace/odoo/odoo/addons/l10n_it_fatturapa_in/tests/test_import_fatturapa_xml.py", line 197, in test_04_xml_import
: `     u"Company Name field contains 'Societa\' "
: ` AssertionError: u'Il campo nome azienda contiene "Societa\' Alpha SRL", il sistema contiene "SOC [truncated]... != u"Company Name field contains 'Societa' Alpha SRL'. Your System contains 'SOCIET [truncated]...
: ` - Il campo nome azienda contiene "Societa' Alpha SRL", il sistema contiene "SOCIETA' ALPHA SRL"
: ` + Company Name field contains 'Societa' Alpha SRL'. Your System contains 'SOCIETA' ALPHA SRL'
: `   
: ` - L'imponibile calcolato 25.0 \xe8 diverso da quello dei dati di riepilogo 26.0+ Computed amount untaxed 25.0 is different from summary data 26.0


FAIL: test_09_xml_import (odoo.addons.l10n_it_fatturapa_in.tests.test_import_fatturapa_xml.TestFatturaPAXMLValidation)
: Traceback (most recent call last):
: `   File "/home/sherpya/workspace/odoo/odoo/addons/l10n_it_fatturapa_in/tests/test_import_fatturapa_xml.py", line 263, in test_09_xml_import
: `     'Computed amount untaxed 1030.42 is different from'
: ` AssertionError: u"L'imponibile calcolato 1030.42 \xe8 diverso da quello dei dati di riepilogo 1173.6" != 'Computed amount untaxed 1030.42 is different from summary data 1173.6'

FAIL: test_13_xml_import (odoo.addons.l10n_it_fatturapa_in.tests.test_import_fatturapa_xml.TestFatturaPAXMLValidation)
: Traceback (most recent call last):
: `   File "/home/sherpya/workspace/odoo/odoo/addons/l10n_it_fatturapa_in/tests/test_import_fatturapa_xml.py", line 321, in test_13_xml_import
: `     u"Company Name field contains 'Societa\' "
: ` AssertionError: u'Il campo nome azienda contiene "Societa\' Alpha SRL", il sistema contiene "SOC [truncated]... != u"Company Name field contains 'Societa' Alpha SRL'. Your System contains 'SOCIET [truncated]...
: ` - Il campo nome azienda contiene "Societa' Alpha SRL", il sistema contiene "SOCIETA' ALPHA SRL"
: ` + Company Name field contains 'Societa' Alpha SRL'. Your System contains 'SOCIETA' ALPHA SRL'
: `   
: ` - L'imponibile calcolato 25.0 \xe8 diverso da quello dei dati di riepilogo 26.0+ Computed amount untaxed 25.0 is different from summary data 26.0

FAIL: test_14_xml_import (odoo.addons.l10n_it_fatturapa_in.tests.test_import_fatturapa_xml.TestFatturaPAXMLValidation)
: Traceback (most recent call last):
: `   File "/home/sherpya/workspace/odoo/odoo/addons/l10n_it_fatturapa_in/tests/test_import_fatturapa_xml.py", line 344, in test_14_xml_import
: `     u"Company Name field contains 'Societa\' "
: ` AssertionError: u'Il campo nome azienda contiene "Societa\' Alpha SRL", il sistema contiene "SOC [truncated]... != u"Company Name field contains 'Societa' Alpha SRL'. Your System contains 'SOCIET [truncated]...
: ` - Il campo nome azienda contiene "Societa' Alpha SRL", il sistema contiene "SOCIETA' ALPHA SRL"
: ` + Company Name field contains 'Societa' Alpha SRL'. Your System contains 'SOCIETA' ALPHA SRL'
: `   
: ` - Il file XML contiene l'imposta con aliquota "15.55", ma questa non esiste nel sistema.
: ` - Il file XML contiene l'imposta con aliquota "15.55", ma questa non esiste nel sistema.+ XML contains tax with percentage '15.55' but it does not exist in your system 
: ` + XML contains tax with percentage '15.55' but it does not exist in your system
```

--- a/l10n_it_fatturapa_in/__manifest__.py
+++ b/l10n_it_fatturapa_in/__manifest__.py
@@ -6,7 +6,7 @@
 
 {
     'name': 'Italian Localization - Fattura elettronica - Ricezione',
-    'version': '12.0.1.1.1',
+    'version': '12.0.1.1.2',
     "development_status": "Beta",
     'category': 'Localization/Italy',
     'summary': 'Ricezione fatture elettroniche',

--- a/l10n_it_fatturapa_in/tests/data/IT05979361218_010.xml
+++ b/l10n_it_fatturapa_in/tests/data/IT05979361218_010.xml
@@ -1,0 +1,101 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<p:FatturaElettronica versione="FPA12" xmlns:ds="http://www.w3.org/2000/09/xmldsig#"
+xmlns:p="http://ivaservizi.agenziaentrate.gov.it/docs/xsd/fatture/v1.2"
+xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+xsi:schemaLocation="http://ivaservizi.agenziaentrate.gov.it/docs/xsd/fatture/v1.2 http://www.fatturapa.gov.it/export/fatturazione/sdi/fatturapa/v1.2/Schema_del_file_xml_FatturaPA_versione_1.2.xsd">
+  <FatturaElettronicaHeader>
+    <DatiTrasmissione>
+      <IdTrasmittente>
+        <IdPaese>IT</IdPaese>
+        <IdCodice>05979361218</IdCodice>
+      </IdTrasmittente>
+      <ProgressivoInvio>009</ProgressivoInvio>
+      <FormatoTrasmissione>FPA12</FormatoTrasmissione>
+      <CodiceDestinatario>UFPQ1O</CodiceDestinatario>
+      <ContattiTrasmittente/>
+    </DatiTrasmissione>
+    <CedentePrestatore>
+      <DatiAnagrafici>
+        <IdFiscaleIVA>
+            <IdPaese>IT</IdPaese>
+            <IdCodice>05979361218</IdCodice>
+        </IdFiscaleIVA>
+        <Anagrafica>
+          <Denominazione>SOCIETA' ALPHA SRL</Denominazione>
+        </Anagrafica>
+        <RegimeFiscale>RF19</RegimeFiscale>
+      </DatiAnagrafici>
+      <Sede>
+        <Indirizzo>VIALE ROMA 543</Indirizzo>
+        <CAP>07100</CAP>
+        <Comune>SASSARI</Comune>
+        <Provincia>SS</Provincia>
+        <Nazione>IT</Nazione>
+      </Sede>
+    </CedentePrestatore>
+    <CessionarioCommittente>
+      <DatiAnagrafici>
+        <CodiceFiscale>80213330584</CodiceFiscale>
+        <Anagrafica>
+          <Denominazione>DITTA BETA</Denominazione>
+        </Anagrafica>
+      </DatiAnagrafici>
+      <Sede>
+        <Indirizzo>VIA TORINO 38-B</Indirizzo>
+        <CAP>00145</CAP>
+        <Comune>ROMA</Comune>
+        <Provincia>RM</Provincia>
+        <Nazione>IT</Nazione>
+      </Sede>
+    </CessionarioCommittente>
+  </FatturaElettronicaHeader>
+  <FatturaElettronicaBody>
+    <DatiGenerali>
+      <DatiGeneraliDocumento>
+        <TipoDocumento>TD01</TipoDocumento>
+        <Divisa>EUR</Divisa>
+        <Data>2014-12-18</Data>
+        <Numero>129</Numero>
+        <DatiRitenuta>
+            <TipoRitenuta>RT01</TipoRitenuta>
+            <ImportoRitenuta>1.00</ImportoRitenuta>
+            <AliquotaRitenuta>20.00</AliquotaRitenuta>
+            <CausalePagamento>A</CausalePagamento>
+        </DatiRitenuta>
+        <Causale>LA FATTURA FA RIFERIMENTO AD UNA OPERAZIONE AAAA BBBBBBBBBBBBBBBBBB CCC DDDDDDDDDDDDDDD E FFFFFFFFFFFFFFFFFFFF GGGGGGGGGG HHHHHHH II LLLLLLLLLLLLLLLLL MMM NNNNN OO PPPPPPPPPPP QQQQ RRRR SSSSSSSSSSSSSS</Causale>
+        <Causale>SEGUE DESCRIZIONE CAUSALE NEL CASO IN CUI NON SIANO STATI SUFFICIENTI 200 CARATTERI AAAAAAAAAAA BBBBBBBBBBBBBBBBB</Causale>
+      </DatiGeneraliDocumento>
+      <DatiFattureCollegate>
+        <RiferimentoNumeroLinea>1</RiferimentoNumeroLinea>
+        <IdDocumento>1/PF</IdDocumento>
+        <Data>2014-12-18</Data>
+        <NumItem>1</NumItem>
+      </DatiFattureCollegate>
+    </DatiGenerali>
+    <DatiBeniServizi>
+      <DettaglioLinee>
+        <NumeroLinea>1</NumeroLinea>
+        <Descrizione>DESCRIZIONE DELLA FORNITURA</Descrizione>
+        <Quantita>5.00</Quantita>
+        <PrezzoUnitario>1.00</PrezzoUnitario>
+        <PrezzoTotale>5.00</PrezzoTotale>
+        <AliquotaIVA>22.00</AliquotaIVA>
+        <Ritenuta>SI</Ritenuta>
+      </DettaglioLinee>
+      <DatiRiepilogo>
+        <AliquotaIVA>22.00</AliquotaIVA>
+        <ImponibileImporto>5.00</ImponibileImporto>
+        <Imposta>1.10</Imposta>
+        <EsigibilitaIVA>I</EsigibilitaIVA>
+      </DatiRiepilogo>
+    </DatiBeniServizi>
+    <DatiPagamento>
+      <CondizioniPagamento>TP01</CondizioniPagamento>
+      <DettaglioPagamento>
+        <ModalitaPagamento>MP01</ModalitaPagamento>
+        <DataScadenzaPagamento>2015-01-30</DataScadenzaPagamento>
+        <ImportoPagamento>5.10</ImportoPagamento>
+      </DettaglioPagamento>
+    </DatiPagamento>
+  </FatturaElettronicaBody>
+</p:FatturaElettronica>

--- a/l10n_it_fatturapa_in/tests/test_import_fatturapa_xml.py
+++ b/l10n_it_fatturapa_in/tests/test_import_fatturapa_xml.py
@@ -416,3 +416,10 @@ class TestFatturaPAXMLValidation(SingleTransactionCase):
                 self.assertTrue(len(invoice.invoice_line_ids) == 2)
             if invoice.reference == '456':
                 self.assertTrue(len(invoice.invoice_line_ids) == 1)
+
+    def test_17_xml_import(self):
+        res = self.run_wizard('test17', 'IT05979361218_010.xml')
+        invoice_id = res.get('domain')[0][2][0]
+        invoice = self.invoice_model.browse(invoice_id)
+        self.assertEqual(
+            invoice.related_documents[0].type, "invoice")

--- a/l10n_it_fatturapa_in/wizard/wizard_import_fatturapa.py
+++ b/l10n_it_fatturapa_in/wizard/wizard_import_fatturapa.py
@@ -964,9 +964,9 @@ class WizardImportFatturapa(models.TransientModel):
         # 2.1.6
         RelInvoices = FatturaBody.DatiGenerali.DatiFattureCollegate
         if RelInvoices:
-            for invoice in RelInvoices:
+            for rel_invoice in RelInvoices:
                 doc_datas = self._prepareRelDocsLine(
-                    invoice_id, invoice, 'invoice')
+                    invoice_id, rel_invoice, 'invoice')
                 if doc_datas:
                     for doc_data in doc_datas:
                         rel_docs_model.create(doc_data)

